src/adaptive_exp_and_gram.jl

```diff
@@ -1,72 +1,68 @@
 """
     AdaptiveExpAndGram{T,A} <: AbstractExpAndGramAlgorithm
 
-Adaptive algorithm for computing the matrix exponential and an associated Gramian. 
+Adaptive algorithm for computing the matrix exponential and an associated Gramian.
 
-Constructor: 
+Constructor:
 
 AdaptiveExpAndGram{T}()
 
-creates an algorithm with coefficients stored in the numeric type T. 
+creates an algorithm with coefficients stored in the numeric type T.
 
 """
 struct AdaptiveExpAndGram{T,A} <: AbstractExpAndGramAlgorithm where {A}
     methods::A
+    function AdaptiveExpAndGram{T}() where {T}
+        methods = [ExpAndGram{T,q}() for q in (3, 5, 7, 9, 13)]
+        #normtols = T[0.00067, 0.021, 0.13, 0.41, 1.57] # normtols for Gramian
+        #normtols = T[0.0014, 0.25, 0.95, 2.1, 5.39] # normtols for matrix exponential
+        return new{T,typeof(methods)}(methods)
+    end
 end
 
-function AdaptiveExpAndGram{T}() where {T}
-    methods = [ExpAndGram{T,q}() for q in (3, 5, 7, 9, 13)]
-    #normtols = T[0.00067, 0.021, 0.13, 0.41, 1.57] # normtols for Gramian 
-    #normtols = T[0.0014, 0.25, 0.95, 2.1, 5.39] # normtols for matrix exponential 
-    return AdaptiveExpAndGram{T,typeof(methods)}(methods)
-end
-
-function exp_and_gram(
-    A::AbstractMatrix{T},
-    B::AbstractMatrix{T},
-    method::AdaptiveExpAndGram,
-) where {T<:Number}
-    Φ, G = exp_and_gram!(copy(A), copy(B), method)
-    return Φ, G
-end
 
-function exp_and_gram!(
-    A::AbstractMatrix{T},
-    B::AbstractMatrix{T},
-    method::AdaptiveExpAndGram,
-) where {T<:Number}
-    Φ, U = exp_and_gram_chol!(A, B, method)
-    G = U' * U
-    _symmetrize!(G)
-    return Φ, G
+function adaptive_to_alg(A, B, method::AdaptiveExpAndGram)
```

Comment:
Initially wanted to replace the logic below with this, until I realized that in one case it calls `_exp_and_gram_chol_init`, in the other it's `exp_and_gram_chol!` :/